airbyte-integrations/connectors/source-netsuite/source_netsuite/streams.py

```diff
@@ -49,9 +51,17 @@ def __init__(
 
     raise_on_http_errors = True
 
+    @property
+    def get_input_date_formats(self) -> List[str]:
+        if self.default_date_format is not None:
+            if self.default_date_format not in NETSUITE_INPUT_DATE_FORMATS:
```

Comment:
nitpick: You can roll the inner condition into the original check with an `and`.